lib/fs.js

```diff
@@ -569,16 +609,40 @@ function open(path, flags, mode, callback) {
  * @param {string | number} [flags]
  * @param {string | number} [mode]
  * @returns {number}
+ *
+ * or
+ *
+ * @param {string | Buffer | URL} path
+ * @param {{
+ *   flags?: string | number;
+ *   mode?: string | number;
+ *   } | string} [options]
+ * @returns {number}
  */
 function openSync(path, flags, mode) {
+  let options;
+  let parents;
   path = getValidatedPath(path);
-  const flagsNumber = stringToFlags(flags);
+  // Handle case where 2nd argument is options-bag
+  if (typeof flags === 'object' && flags) {
```

Comment:
📓 similar to https://github.com/nodejs/node/pull/35775/files#r1380873544
```suggestion
  if (typeof flags === 'object') {
```